backend/src/main/java/ca/gov/dtsstn/cdcp/api/service/SubscriptionService.java

```diff
@@ -8,32 +9,53 @@
 import org.springframework.util.Assert;
 
 import ca.gov.dtsstn.cdcp.api.data.repository.SubscriptionRepository;
+import ca.gov.dtsstn.cdcp.api.data.repository.UserRepository;
 import ca.gov.dtsstn.cdcp.api.service.domain.Subscription;
 import ca.gov.dtsstn.cdcp.api.service.domain.mapper.SubscriptionMapper;
 
 @Service
 public class SubscriptionService {
 
-	private final SubscriptionMapper subscriptionMapper = Mappers.getMapper(SubscriptionMapper.class);
-
 	private final SubscriptionRepository subscriptionRepository;
 
-	public SubscriptionService(SubscriptionRepository repository) {
-		Assert.notNull(repository, "subscriptionRepository is required; it must not be null");
-		this.subscriptionRepository = repository;
+	SubscriptionMapper subscriptionMapper = Mappers.getMapper(SubscriptionMapper.class);
+
+	private final UserRepository userRepository;
+
+	public SubscriptionService(
+				SubscriptionRepository subscriptionRepository, 
+				UserRepository userRepository) {
+		Assert.notNull(subscriptionRepository, "subscriptionRepository is required; it must not be null");
+		Assert.notNull(userRepository, "userRepository is required; it must not be null");
+		this.subscriptionRepository = subscriptionRepository;
+		this.userRepository = userRepository;
 	}
 
-	public Subscription create(Subscription subscription) {
+	public void create(Subscription subscription, String userId) {
 		Assert.notNull(subscription, "subscription is required; it must not be null");
-		Assert.isNull(subscription.getId(), "subscription.id must be null when creating new instance");
-
-		return subscriptionMapper.fromEntity(subscriptionRepository.save(subscriptionMapper.toEntity(subscription)));
+		Assert.notNull(userId, "userId is required; it must not be null");
+		final var userEntity = userRepository.findById(userId).get();
+		final var subscriptions = userEntity.getSubscriptions();
+		if (subscriptions.size() == 0) {
+			userEntity.setSubscriptions(Arrays.asList(subscriptionMapper.toEntity(subscription)));				
```

Comment:
Since `subscriptions` should be an empty set if no subscriptions exist, there's no need to initially check if `subscriptions` is empty - we can just add the new subscription to the existing (empty) set.